app/Http/Controllers/Front/ClientController.php

```diff
@@ -890,4 +892,43 @@ public function getPaymentByOrderIdClient($orderid, $userid)
             return redirect()->back()->with('fails', $ex->getMessage());
         }
     }
+
+    /**
+     * Delete an invoice and its related records based on specific conditions.
+     *
+     * @param  int  $id  The ID of the invoice to be deleted.
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function invoiceDelete($id)
```

Comment:
public function invoiceDelete($id)
{
    $invoice = Invoice::find($id);

    if (! $invoice) {
        return response()->json(['error' => 'Invoice not found'], 404);
    }

    if ($this->canDeleteInvoice($invoice)) {
        $this->deleteInvoice($invoice);

        return response()->json(['message' => 'Invoice deleted successfully']);
    }

    return response()->json(['error' => 'Cannot delete invoice.'], 400);
}

private function canDeleteInvoice($invoice)
{
    return (
        $invoice->is_renewed == 0 &&
        ! $invoice->orderRelation()->exists() &&
        $invoice->invoiceItem()->exists()
    ) || (
        $invoice->is_renewed != 0 &&
        $invoice->orderRelation()->exists() &&
        $invoice->invoiceItem()->exists()
    );
}

private function deleteInvoice($invoice)
{
    $invoice->invoiceItem()->delete();

    if ($invoice->is_renewed != 0 && $invoice->orderRelation()->exists()) {
        $invoice->orderRelation()->delete();
    }

    $invoice->delete();
    \Session::forget('invoice');
}
 

This can be split like this